src/dom/document.ts

```diff
@@ -307,8 +307,25 @@ export class Document extends Node {
   private _getElementsByClassName(className: string, search: Node[]): Node[] {
     for (const child of this.childNodes) {
       if (child.nodeType === NodeType.ELEMENT_NODE) {
-        if ((<Element> child).classList.contains(className)) {
-          search.push(child);
+        // remove duplicate class names
+        const classList = new Set(className.split(" "));
```

Comment:
I don't think that it's worth creating a `Set` here, it's basically offset by `.classList.contains` internally invoking `Set.has` thus basically doing this work for us with an already created `Set`. Also, you want to split by adjacent whitespace, i.e.
```
foo  bar
```
with two spaces incorrectly counts as three classnames.